utopia-remix/app/services/fgaClient.server.ts

```diff
@@ -0,0 +1,65 @@
+import { CredentialsMethod, ErrorCode, FgaApiValidationError, OpenFgaClient } from '@openfga/sdk'
+import { ServerEnvironment } from '../env.server'
+
+const mockOpenFgaClient = {
+  write: async () => {},
+  check: async () => ({ allowed: true }),
+}
+
+export const fgaClient = createFgaClient()
+
+export function createFgaClient() {
+  const fgaClientMode = getFgaClientMode()
+  if (fgaClientMode === 'local_mock') {
+    return mockOpenFgaClient
+  }
+  return new OpenFgaClient({
+    apiScheme: ServerEnvironment.FGA_API_SCHEME,
+    apiHost: ServerEnvironment.FGA_API_HOST,
+    storeId: ServerEnvironment.FGA_STORE_ID,
+    // authorizationModelId: 'YOUR_MODEL_ID', // optional
+    credentials: getCredentials(fgaClientMode),
+  })
+}
+
+function getCredentials(fgaClientMode: 'local' | 'remote') {
+  // if working against a local server, we don't need credentials
+  if (fgaClientMode === 'local') {
+    return { method: CredentialsMethod.None } as const
```

Comment:
Would it not make sense to define the return type and then I don't believe you'd need the `as const` suffixes?

Reply:
The problem is that openfga doesn't export the return type at all. I can just define it to be a union, though